### PR TITLE
KAFKA-14587: Move AclCommand to tools

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/test/java/org/apache/kafka/tools/AclCommandTest.java [hunk 1]
```diff
@@ -303,7 +294,7 @@ public void testUseBootstrapServerOptWithBootstrapControllerOpt() {
     @Test
     public void testUseWithoutBootstrapServerOptAndBootstrapControllerOpt() {
         assertInitializeInvalidOptionsExitCodeAndMsg(
-                Collections.emptyList(),
+                Collections.singletonList(ADD),
```
Comment (mimaison) [inline tools/src/test/java/org/apache/kafka/tools/AclCommandTest.java]: Actually if you don't provide any arguments on the CLI, just the help message is printed.

The previous test was expecting a different behavior as `maybePrintHelpOrVersion()` used to be called in `main()` while I moved it to `AclCommandOptions.checkArgs()` for consistency with the other tools, and consistency with the CLI behavior.

File: tools/src/test/java/org/apache/kafka/tools/AclCommandTest.java [hunk 2]
```diff
@@ -355,7 +346,7 @@ public void testInvalidArgs() {
                 "Option \"[list]\" can't be used with option \"[producer]\""
         );
         assertInitializeInvalidOptionsExitCodeAndMsg(
-                Arrays.asList(BOOTSTRAP_SERVER, LOCALHOST, ADD, PRODUCER, OPERATION),
+                Arrays.asList(BOOTSTRAP_SERVER, LOCALHOST, ADD, PRODUCER, OPERATION, "all"),
```
Comment (mimaison) [inline tools/src/test/java/org/apache/kafka/tools/AclCommandTest.java]: Running the same command on the CLI did not produce that error message, so this test was effectively testing internal logic that would not be reachable for users.
```
./bin/kafka-acls.sh --bootstrap-server localhost:9092 --add --producer --operation
Exception in thread "main" joptsimple.OptionMissingRequiredArgumentException: Option operation requires an argument
```
To produce that error message from the CLI you need to specify an operation.

**Comment by mimaison** [comment]
@chia7712 @showuon Can you take a look if you have time? Thanks

**Review by chia7712 — APPROVED** [review]

**Comment by chia7712** [comment]
@mimaison could you please fix conflicts?

**Pull request merged**